homeassistant/components/intent/__init__.py

```diff
@@ -70,24 +72,29 @@ async def async_call_service(self, intent_obj: intent.Intent, state: State) -> N
         if state.domain == COVER_DOMAIN:
             # on = open
             # off = close
-            await hass.services.async_call(
-                COVER_DOMAIN,
-                SERVICE_OPEN_COVER
-                if self.service == SERVICE_TURN_ON
-                else SERVICE_CLOSE_COVER,
-                {ATTR_ENTITY_ID: state.entity_id},
-                context=intent_obj.context,
-                blocking=True,
-                limit=self.service_timeout,
+            task = hass.async_create_task(
+                hass.services.async_call(
+                    COVER_DOMAIN,
+                    SERVICE_OPEN_COVER
+                    if self.service == SERVICE_TURN_ON
+                    else SERVICE_CLOSE_COVER,
+                    {ATTR_ENTITY_ID: state.entity_id},
+                    context=intent_obj.context,
+                    blocking=True,
+                )
             )
+            # Block with a short timeout to (hopefully) catch validation errors
+            with contextlib.suppress(asyncio.TimeoutError):
+                await asyncio.wait_for(task, timeout=self.service_timeout)
```

Comment:
`wait_for` will cancel the task on timeout.

Reply:
Addressed + added test for it.